src/components/NewMessage/NewMessage.vue

```diff
@@ -595,6 +655,20 @@ export default {
 			}
 		},
 
+		async handleEdit() {
+			try {
+				await editMessage({
+					token: this.token,
+					messageId: this.messageToEdit.id,
+					updatedMessage: this.text.trim()
+				})
+				EventBus.$emit('message-edited', this.messageToEdit.id)
```

Comment:
~~Make sure to emit something in case of fail, to switch on drag-n-drop again ~~

UPD: I'd rely on a store here